sw/device/lib/crypto/drivers/kmac.h

```diff
@@ -148,17 +161,22 @@ kmac_error_t kmac_write_prefix_block(kmac_operation_t operation,
                                      crypto_const_uint8_buf_t cust_str);
 
 /**
- * Update the key registers with given key.
+ * Update the key registers with given key shares.
+ *
+ * The caller must ensure that `key_share0` and `key_share1` are not NULL
+ * pointers. This is not checked within this function.
  *
- * The caller must ensure that `key` is not a NULL pointer. This is not checked
- * within this function.
+ * The accepted `key_len` values are {128 / 8, 192 / 8, 256 / 8, 384 / 8,
+ * 512 / 8}, otherwise an error will be returned.
  *
- * @param key The input key array.
- * @param key_len The size of key from enum type kmac_key_len_t.
+ * @param key_share0 The first share of the key array.
+ * @param key_share1 The second share of the key array.
+ * @param key_len The size of each share in bytes.
  * @return Error code.
  */
 OT_WARN_UNUSED_RESULT
-kmac_error_t kmac_write_key_block(const uint8_t *key, kmac_key_len_t key_len);
+kmac_error_t kmac_write_key_block(const uint32_t *key_share0,
+                                  const uint32_t *key_share1, size_t key_len);
```

Comment:
I think it might make sense to define a struct here (which I will do shortly), but I would like to still keep key pointers as `uint32_t*` and the length in byte units:

```
typedef struct word_aligned_two_share_key {
  uint32_t *share0;
  uint32_t *share1;
  // `key_len` is the byte size of single share
  size_t key_len;
}
```
This would ensure compatibility with `crypto_blinded_key` struct from which we copy these pointers (see `keyblob`):

https://github.com/lowRISC/opentitan/blob/3038e19de16cd36512d8b872c5f2b8a7886f3433/sw/device/lib/crypto/include/datatypes.h#L323-L332

I think using word pointers for passing keys makes more sense, only because it forces us to always access keys at word granularity. We wouldn't want to read/write keys at byte granularity [from SCA perspective](https://github.com/lowRISC/opentitan/issues/16767#issuecomment-1353463320).

In general, in the crypto context, I am more in favor of keeping the length information always with the same unit across the library (and the common unit here for us is byte). Sometimes, the length information might have more use than just indicating the size of the accompanying key pointer. For example, we also use `key_len` to check whether passed key size is valid and also determine an enum value (in function `kmac_get_key_len_bytes` in `kmac.c`).

Reply:
SGTM, thanks for the explanation @ballifatih !